File: frontend/src/types/dailyDecision.ts
export interface WorkoutSuggestion {
  type: string;
  durationMin: number;
  targetTss: number;
  difficulty: string;
  intensityDescription: string;
  description: string;
  indoor: boolean;
}

export interface ConfidenceScore {
  score: number;
  label: string;
  description: string;
}

export interface DecisionReason {
  priority: string;
  signal: string;
  message: string;
  evidence: string;
}

export interface AlternativeOption {
  label: string;
  type: string;
  workout: WorkoutSuggestion;
  rationale: string;
}

export interface DailyDecisionDto {
  decision: 'RIDE' | 'MODIFY' | 'SKIP' | 'INDOOR';
  workout: WorkoutSuggestion;
  confidence: ConfidenceScore;
  risk: 'LOW' | 'MODERATE' | 'HIGH' | 'CRITICAL';
  reasons: DecisionReason[];
  alternatives: AlternativeOption[];
}

File: backend/src/main/java/pl/strava/analizator/domain/model/DailyDecision.java
package pl.strava.analizator.domain.model;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class DailyDecision {
    private final DecisionType decision;
    private final WorkoutSuggestion workout;
    private final ConfidenceScore confidence;
    private final RiskLevel risk;
    private final List<DecisionReason> reasons;
    private final List<AlternativeOption> alternatives;
}
